sdk/typescript/src/transactions/Transaction.ts

```diff
@@ -98,12 +98,38 @@ export function isTransaction(obj: unknown): obj is Transaction {
 
 export type TransactionObjectInput = string | CallArg | TransactionObjectArgument;
 
+const TRANSACTION_REGISTRY_KEY = Symbol.for('@mysten/transaction/registry');
+function getGlobalPluginRegistry() {
+	try {
+		const target = globalThis as {
+			[TRANSACTION_REGISTRY_KEY]?: {
+				buildPlugins: TransactionPlugin[];
+				serializationPlugins: TransactionPlugin[];
+			};
+		};
+
+		if (!target[TRANSACTION_REGISTRY_KEY]) {
+			target[TRANSACTION_REGISTRY_KEY] = {
+				buildPlugins: [],
+				serializationPlugins: [],
+			};
+		}
+
+		return target[TRANSACTION_REGISTRY_KEY];
+	} catch (e) {
+		return {
```

Comment:
Could this be a module-level variable instead as the fallback so that it's at least consistent between function calls here?